langchain4j-vertex-ai/src/main/java/dev/langchain4j/model/vertexai/internal/MatchingService.java

```diff
@@ -0,0 +1,111 @@
+import lombok.Getter;
+import lombok.NonNull;
+import lombok.extern.slf4j.Slf4j;
+
+import java.util.List;
+import java.util.stream.Collectors;
+
+@Slf4j
+@Builder
+public class MatchingService {
+
+    @NonNull
+    private final String endpoint;
+    @NonNull
+    private final String project;
+    @NonNull
+    private final String location;
+    @NonNull
+    private final String indexEndpointId;
+    @NonNull
+    private final String deployedIndexId;
+    @NonNull
+    private final String indexId;
+    @NonNull
+    private final GcpBlobService gcpBlobService;
+    private final CredentialsProvider credentialsProvider;
+    @Getter
+    @Builder.Default
+    private final boolean returnFullDatapoint = true;
+    @Getter(lazy = true)
+    private final MatchServiceSettings settings = initSettings();
+
+    /**
+     * Add a document to the index
+     *
+     * @param referenceEmbedding the embedding of the document
+     * @param maxResults         the maximum number of results to return
+     * @param minScore           the minimum score to return
+     * @return the list of matching documents
+     */
+    public List<EmbeddingMatch<TextSegment>> findRelevant(Embedding referenceEmbedding, int maxResults, double minScore) {
+        final IndexDatapoint datapoint = IndexDatapoint
+                .newBuilder()
+                .addAllFeatureVector(referenceEmbedding.vectorAsList())
+                .build();
+        final FindNeighborsRequest.Query query = FindNeighborsRequest.Query
+                .newBuilder()
+                .setDatapoint(datapoint)
+                .setNeighborCount(maxResults)
+                .build();
+        final FindNeighborsRequest request = FindNeighborsRequest
+                .newBuilder()
+                .setIndexEndpoint(IndexEndpointName.of(project, location, indexEndpointId).toString())
+                .setDeployedIndexId(deployedIndexId)
+                .addQueries(query)
+                .setReturnFullDatapoint(returnFullDatapoint)
+                .build();
+
+        try (MatchServiceClient client = MatchServiceClient.create(getSettings())) {
+            final FindNeighborsResponse response = client.findNeighbors(request);
+            return response.getNearestNeighborsList()
+                    .stream()
+                    .flatMap(neighbor -> neighbor.getNeighborsList().stream())
+                    .filter(neighbor -> neighbor.getDistance() >= minScore)
+                    .map(neighbor -> {
+                        final String id = neighbor.getDatapoint().getDatapointId();
+                        final String path = "documents/" + id;
+                        final String content = gcpBlobService.download(path);
+                        final VertexAIDocument document = VertexAIDocument.fromJson(content);
+                        final IndexDatapoint resultDatapoint = neighbor.getDatapoint();
+                        return new EmbeddingMatch<>(neighbor.getDistance(),
```

Comment:
it should not be distance, but score

Reply:
VertexAI doesn't return score but a distance.